src/services/smart-fields-values-injector.js

```diff
@@ -33,6 +33,19 @@ function SmartFieldsValuesInjector(record, modelName, fieldsPerModel) {
         if (_.isFunction(value.then)) {
           return value
             .then((result) => {
+              // NOTICE: We need some recursion in order to consider smart fields in smart
+              //         relations. So if the result contains data and the field has a reference
+              //         then we consider we have to inject smart fields within the reference.
+              if (result && result.dataValues && field.reference) {
+                const [referenceModelName] = field.reference.split('.');
+                const smartFieldsValuesInjector = new SmartFieldsValuesInjector(
+                  result,
+                  referenceModelName,
+                  fieldsPerModel,
+                );
+                smartFieldsValuesInjector.perform();
+              }
```

Comment:
[Consistency] You'll also have to use/implement this logic at line 56.
Here is the implementation case of Smart Field `getter` that return a promise (`_.isFunction(value.then)`).

You'll also have to handle this for Smart Field `getter` that return a value IMO.

Reply:
Good catch! Fixed!